pallets/market/src/benchmarking.rs

```diff
@@ -0,0 +1,409 @@
+#![cfg(feature = "runtime-benchmarks")]
+
+use cid::{multihash::Multihash, Cid};
+use codec::Encode;
+use frame_benchmarking::v2::*;
+use frame_support::{pallet_prelude::ConstU32, sp_runtime::BoundedVec, traits::Currency};
+use frame_system::{
+    self,
+    pallet_prelude::{BlockNumberFor, OriginFor},
+    RawOrigin,
+};
+use pallet_storage_provider::Pallet as SpPallet;
+use primitives::{
+    commitment::{FIL_COMMITMENT_UNSEALED, SHA2_256_TRUNC254_PADDED},
+    proofs::RegisteredPoStProof,
+    sector::{builder::SectorPreCommitInfoBuilder, ProveCommitSector, SectorPreCommitInfo},
+    MAX_LABEL_SIZE, MAX_SECTORS_PER_CALL, PEER_ID_MAX_BYTES,
+};
+use rand::{rngs::SmallRng, RngCore, SeedableRng};
+use scale_info::prelude::format;
+use sp_core::{ed25519, hashing::sha2_256};
+use sp_io::crypto::{ed25519_generate, ed25519_sign};
+use sp_runtime::{traits::IdentifyAccount, AccountId32, MultiSignature, MultiSigner};
+use sp_std::{vec, vec::Vec};
+
+use super::*;
+#[allow(unused)]
+use crate::Pallet as MarketPallet;
+
+type BoundedPeerIdBytes = BoundedVec<u8, ConstU32<PEER_ID_MAX_BYTES>>;
+const COLLATERAL: u32 = 10;
+
+fn random_cid<R: RngCore>(rng: &mut R) -> Cid {
+    let mut buffer = [0u8; 32];
+    rng.fill_bytes(&mut buffer);
+
+    let hash = sha2_256(&buffer);
+    let mh = Multihash::<64>::wrap(SHA2_256_TRUNC254_PADDED, &hash).unwrap();
+    Cid::new_v1(FIL_COMMITMENT_UNSEALED, mh)
+}
```

Comment:
```suggestion
fn commp() -> Cid {
    Commitment::<CommP>::zero(PaddedPieceSize::MIN).cid()
}

```

I don't see a reason to generate it randomly.

Reply:
Because the deals will all be the same and generate conflicts

Reply:
They can be different just by changing labels, indexing by bytes etc.

Reply:
Your call, not major.